test/test_embedded.py

```diff
@@ -0,0 +1,60 @@
+import unittest
+import time
+import os
+import signal
+from unittest.mock import patch
+import pathlib
+
+from weaviate import embedded
+from weaviate.embedded import EmbeddedDB
+from weaviate.exceptions import WeaviateStartUpError
+
+
+class TestEmbeddedBasics(unittest.TestCase):
+    def test__init__(self):
+        self.assertEqual(EmbeddedDB(url="embedded").port, 6666)
+
+    def test__init__non_default_port(self):
+        self.assertEqual(EmbeddedDB(url="embedded?port=30666").port, 30666)
+
+
+class TestEmbeddedEndToEnd(unittest.TestCase):
+    def setUp(self) -> None:
+        embedded.weaviate_binary_path = "./weaviate-embedded-unitests-delete-me"
+        self.embedded_db: EmbeddedDB = None
+
+    def tearDown(self) -> None:
+        file = pathlib.Path(embedded.weaviate_binary_path)
+        if file.exists():
+            file.unlink()
+        self.embedded_db.stop()
+
+    def test_end_to_end_all(self):
```

Comment:
instead of having a loop in the test you can do [parametrization](https://docs.pytest.org/en/6.2.x/parametrize.html)

Reply:
done after changing to pytest